src/server.rs

```diff
@@ -717,46 +682,42 @@ fn handle_post_action(
     state: web::Data<AppState>,
     body: web::Json<serde_json::Value>,
 ) -> HttpResponse {
-    let thing = state.get_thing(req.match_info().get("thing_id"));
-    if thing.is_none() {
+    let thing = if let Some(thing) = state.get_thing(req.match_info().get("thing_id")) {
+        thing
+    } else {
         return HttpResponse::NotFound().finish();
-    }
+    };
 
-    let thing = thing.unwrap();
-
-    if !body.is_object() {
-        return HttpResponse::BadRequest().finish();
-    }
-
-    let action_name = req.match_info().get("action_name");
-    if action_name.is_none() {
+    let action_name = if let Some(action_name) = req.match_info().get("action_name") {
+        action_name
+    } else {
         return HttpResponse::NotFound().finish();
-    }
+    };
 
-    let action_name = action_name.unwrap();
-
-    let message = body.as_object().unwrap();
+    let message = if let Some(message) = body.as_object() {
+        message
+    } else {
+        return HttpResponse::BadRequest().finish();
+    };
 
     let keys: Vec<&String> = message.keys().collect();
     if keys.len() != 1 {
         return HttpResponse::BadRequest().finish();
     }
 
-    if keys[0] != action_name {
+    let input = if let Some(action_params) = message.get(action_name) {
```

Comment:
```suggestion
    if keys[0] != action_name {
        return HttpResponse::BadRequest().finish();
    }
    
    let input = if let Some(action_params) = message.get(action_name) {
```
This one got deleted, but It seems important to me🤔

Reply:
`keys` is `message.keys()`, so `keys[0] != action_name` is already asserted by the fact that we check `keys.len() == 1` and get the key with `message.get(action_name)`.